platform-controller/src/main/java/org/hobbit/controller/ExperimentManager.java

```diff
@@ -146,9 +150,17 @@ public void createNextExperiment() {
             return;
         }
         try {
-            // if there is no benchmark running (and the queue has been
-            // initialized)
-            if ((experimentStatus == null) && (controller.queue != null)) {
+            // if there is no benchmark running, the queue has been
+            // initialized and cluster is healthy
+            ClusterManager clusterManager = new ClusterManagerImpl();
+            boolean isClusterHealthy = clusterManager.isClusterHealthy();
+            if(!isClusterHealthy) {
+                LOGGER.error("Can not start next experiment in the queue, cluster is NOT HEALTHY. " +
+                        "Check your cluster consistency or adjust SWARM_NODE_NUMBER environment variable." +
+                        "Expected number of nodes: "+clusterManager.getExpectedNumberOfNodes()+
+                        "Current number of nodes: "+clusterManager.getNumberOfNodes());
+            }
+            if ((experimentStatus == null) && (controller.queue != null) && (isClusterHealthy)) {
```

Comment:
I would like to move the health check inside this `if` statement, i.e.,
```java
if ((experimentStatus == null) && (controller.queue != null)) {
    // do the health check here...
    if(!isClusterHealthy) {
        // print error
        return;
    }
```
With that structure, the manager is not checking the health of the cluster just to realize that it is not really interested in the health because there is an experiment running or there is no experiment to start. But this might be up to discussion.